name: github-actions-test ci
on:
  pull_request:
    types: [ready_for_review]
  push: 
  workflow_dispatch:

jobs:
  test:
    name: Checkout the E2E Repository and Run tests
    runs-on: ubuntu-latest
    steps:

      ##Found this fix here: https://github.com/actions/checkout/issues/14#issuecomment-523916396
      - name: Fix up git URLs
        run: echo -e '[url "git@github.com:"]\n  insteadOf = "https://github.com/"' >> ~/.gitconfig

      - name: Checkout Main Repo
        uses: actions/checkout@v2
        with:
          ref: chore/sc-7778/if-a-test-fails-in-the-e2e-repo-it-gets-reflected 

      - name: Check out E2E Repo
        uses: actions/checkout@v2
        with:
          repository: theartofeducation/aoeu-e2e
          ref: refs/heads/main
          token: ${{ secrets.ACCESS_TOKEN }}
          persist-credentials: false
          ssh-key: ${{ secrets.PERSONAL_TOKEN }}
          path: actions/workflows/ci.yml
   
      - name: Run my action
        uses: ./.github/workflows/ci.yml
        with:
          token: ${{ secrets.ACCESS_TOKEN }}
          ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}

